lib/utility/runJoin.js

```diff
@@ -82,42 +82,44 @@ exports.runJoin = function runJoin(Model, isCollection, instances, models, isInn
 			}
 			return next();
 		}
+		self.logger.debug('Querying ' + model.name + ' where ' + JSON.stringify(query));
 		self.execModelMethod(Model, model, 'query', {
 			where: query,
 			limit: calculateLimit(model)
 		}, function queryCallback(err, result) {
-			if (!retVal) { return next(); }
 			if (err) { return next(err); }
-			if (model.multiple) {
-				if (result && result.length) {
-					for (var i = 0; i < result.length; i++) {
-						var item = result[i].toJSON();
-						// if its join as "field" then extract just the field value, else the complete set
-						// if it's a join as field it will always have a name property which we can use to get the
-						// field name from the fetched result.
-						var originalFieldName = Model.fields[model.fieldName].name;
-						item = model.mergeType === 'field' ? item[originalFieldName] : item;
-						if (!retVal[model.fieldName]) {
-							retVal[model.fieldName] = [item];
-						}
-						else {
-							retVal[model.fieldName].push(item);
-						}
-					}
-				}
-				else if (isInnerJoin) {
-					retVal = null;
-				}
+			self.logger.debug('Performing', isInnerJoin ? 'inner join' : 'left join');
+			self.logger.debug('result', result);
+			if ((!result || !result.length) && isInnerJoin) { 
+				retVal = null;
 			}
-			else {
-				if (result) {
-					self.populateValuesFromResult(Model, model, retVal, result);
+			else if (model.multiple) {
+				// since the current field in the main model may have been aliased, 
+				// we should get the name of the field being referred to in the joined model.
+				var key = Model.fields[model.fieldName].name;
+
+				// initialise the array to be returned.
+				// in the future if we actually want to return an empty array when nothing matched we can
+				// remove this
+				if (result.length) {
+					retVal[model.fieldName] = [];
 				}
-				else if (isInnerJoin) {
-					retVal = null;
+				// loop through each result and build the array to be returned
+				for (var i = 0; i < result.length; i++) {
```

Comment:
Not sure why i did it like that. I'm guessing because of the comment above if(result.length) so the if statement could easily be removed.

I'll move it back in.